### PR TITLE
support for `cancel_on`

### PR DESCRIPTION
**Pull request by buixor**
- Support for `cancel_on` in buckets
 - [ ] document in crowdsec-docs

### CODE REVIEW

**Review by sabban — APPROVED** [review]
lgtm

File: pkg/leakybucket/bucket.go [hunk 1]
```diff
@@ -237,7 +239,15 @@ func LeakRoutine(leaky *Leaky) error {
 		case ofw := <-leaky.Out:
 			leaky.overflow(ofw)
 			return nil
-			/*we underflow or reach bucket deadline (timers)*/
+		/*suiciiiide*/
+		case <-leaky.Suicide:
+			close(leaky.Signal)
+			BucketsUnderflow.With(prometheus.Labels{"name": leaky.Name}).Inc()
```
Comment (AlteredCoder) [inline pkg/leakybucket/bucket.go:245]: Should we considered this as underflow ? don't we want a `BucketCanceled` metric ?

File: pkg/leakybucket/manager_load.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ type BucketFactory struct {
 	RunTimeGroupBy  *vm.Program               `json:"-"`
 	Data            []*types.DataSource       `yaml:"data,omitempty"`
 	DataDir         string                    `yaml:"-"`
+	ResetFilter     string                    `yaml:"cancel_on,omitempty"` //a filter that, if matched, kills the bucket
```
Comment (AlteredCoder) [inline pkg/leakybucket/manager_load.go:63]: Shall we have the same property name than the yaml configuration option ?

File: pkg/leakybucket/reset_filter.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+// type : leaky
+// filter: filter: "evt.Meta.log_type == 'http_access-log'
+// reset_filter: evt.Parsed.request endswith '.css'
+// ....
+// Thus, if the bucket receives a request that matches fetching a static ressource (here css), it cancels itself
+
+type ResetFilter struct {
+	ResetFilter          *vm.Program
+	ResetFilterSafeGuard map[string]time.Time
+}
```
Comment (AlteredCoder) [inline pkg/leakybucket/reset_filter.go:24]: Do we want also an `ExprDebugger` for this ?

Comment (cronlabspl) [reply pkg/leakybucket/reset_filter.go:24]: @AlteredCoder isnt `ExportDebugger` attached AT run/compile time?

**Review by cronlabspl — CHANGES_REQUESTED** [review]

**Review by AlteredCoder — APPROVED** [review]

**Pull request merged**